### PR TITLE
ida-free: init at 8.2.230124

### PR DESCRIPTION
**Pull request by bbjubjub2494**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/analysis/ida-free/default.nix [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+  meta = {
+    description = "Freeware version of the world's smartest and most feature-full disassembler";
+    homepage = "https://hex-rays.com/ida-free/";
+    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
+    license = lib.licenses.unfree;
+    maintainers = with lib.maintainers; [ lourkeur ];
+    platforms = [ "x86_64-linux" ];
+    mainProgram = "ida64";
+  };
```
Comment (azahi) [inline pkgs/development/tools/analysis/ida-free/default.nix:121]:
```suggestion
  meta = with lib; {
    description = "Freeware version of the world's smartest and most feature-full disassembler";
    homepage = "https://hex-rays.com/ida-free/";
    sourceProvenance = with sourceTypes; [ binaryNativeCode ];
    license = licenses.unfree;
    maintainers = with maintainers; [ lourkeur ];
    platforms = [ "x86_64-linux" ];
    mainProgram = "ida64";
  };
```

File: pkgs/development/tools/analysis/ida-free/update.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i bash -p bash curl jq
+
+set -e
+
+urls=(
+  https://out7.hex-rays.com/files/idafree82_linux.run
+  https://out7.hex-rays.com/files/idafree82_mac.app.zip
+  https://out7.hex-rays.com/files/arm_idafree82_mac.app.zip
+)
```
Comment (azahi) [inline pkgs/development/tools/analysis/ida-free/update.sh:10]: What about version bumps? Will we encounter hash a mismatch here if the upstream is updated?

File: pkgs/development/tools/analysis/ida-free/default.nix [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    exec = "ida64";
+    icon = icon;
+    comment = "Freeware version of the world's smartest and most feature-full disassembler";
+    desktopName = "IDA Free";
+    genericName = "Interactive Disassembler";
+    categories = [ "Development" ];
+  };
+
+  ldLibraryPath = lib.makeLibraryPath [
+    stdenv.cc.cc
```
Comment (azahi) [inline pkgs/development/tools/analysis/ida-free/default.nix:51]: Why is this required?

Comment (lourkeur) [reply pkgs/development/tools/analysis/ida-free/default.nix:51]: `libstdc++.so.6`

File: pkgs/development/tools/analysis/ida-free/default.nix [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    xorg.libXrender
+    xorg.libXau
+    xorg.libX11
+    xorg.libSM
+    xorg.libICE
+    xorg.libXext
+    xorg.xcbutilwm
+    xorg.xcbutilimage
+    xorg.xcbutilkeysyms
+    xorg.xcbutilrenderutil
```
[the remark this answers is not in the document]
Comment (lourkeur) [reply pkgs/development/tools/analysis/ida-free/default.nix:65]: There's no consensus in the existing code base on using `with xorg;`

File: pkgs/development/tools/analysis/ida-free/update.sh (hunk 2 above)
Comment (lourkeur) [reply pkgs/development/tools/analysis/ida-free/update.sh:10]: oh that's correct. I might have to parse https://hex-rays.com/ida-free/#download.

**Comment by lourkeur** [comment]
> When F1 is pressed to open help: image

thank you! should be fixed now

File: pkgs/development/tools/analysis/ida-free/default.nix [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    fontconfig
+    freetype
+    libGL
+  ];
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  dontUnpack = true;
+  dontStrip = true;
+  dontPatchELF = true;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/ida-free/default.nix:82]: Is this breaking things? otherwise please keep them enabled

File: pkgs/development/tools/analysis/ida-free/default.nix [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+  installPhase = ''
+    runHook preInstall
+
+    IDADIR=$out/opt/${pname}
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/ida-free/default.nix:87]: Do we need to use opt?

File: pkgs/development/tools/analysis/ida-free/default.nix [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+      patchelf --set-interpreter $(cat $NIX_CC/nix-support/dynamic-linker) $IDADIR/$bb
+      wrapProgram \
+        $IDADIR/$bb \
+        --set LD_LIBRARY_PATH "$IDADIR:${ldLibraryPath}" \
+        --set QT_PLUGIN_PATH "$IDADIR/plugins/platforms"
+    done
+
+    ln -s $IDADIR/ida64 $out/bin/ida64
+
+    cp $desktopItem/share/applications/* $out/share/applications
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/ida-free/default.nix:108]: Use copy desktop items instead

File: pkgs/development/tools/analysis/ida-free/default.nix [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+let
+  srcs = builtins.fromJSON (builtins.readFile ./srcs.json);
+in
+
+stdenv.mkDerivation rec {
+  pname = "ida-free";
+  version = "8.2.230124";
+
+  src = fetchurl {
+    inherit (srcs.${stdenv.system}) url sha256;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/ida-free/default.nix:30]: this needs a fallback ``or`` or otherwise eval on other platforms breaks

**Comment by Dawoox** [comment]
Is this PR in a state where it could be merge into nixpkgs ?

**Comment by Dawoox** [comment]
Hi, I'm interested to work on this PR, what is the best approach to retrieve the work that has already been done.
Should I fork the fork of the initial maintainer ?

**Comment by bbjubjub2494** [comment]
> Hi, I'm interested to work on this PR, what is the best approach to retrieve the work that has already been done. Should I fork the fork of the initial maintainer ?

Hi! You can but you don't have to do that, you can go to your normal local repository and use `git fetch https://github.com/bbjubjub2494/nixpkgs idafree` and that should give you the commit in `FETCH_HEAD`. Thank you for your willingness.
